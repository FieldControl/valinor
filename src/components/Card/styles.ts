import styled, { keyframes } from 'styled-components';

const scaleIn = keyframes`
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
`;

export const Container = styled.li`
  display: flex;
  align-items: center;
  justify-content: space-between;

  width: 100%;
  border-radius: 5px;
  padding: 1rem;

  background: var(--white);

  animation: ${scaleIn} 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;

  > div {
    display: flex;
    align-items: center;
  }

  > a {
    transition: filter 0.2s;

    &:hover {
      filter: brightness(0.8);
    }
  }

  div img {
    width: 5rem;
    border-radius: 50%;
  }

  div div.info-repo {
    margin-left: 1.5rem;

    a {
      font-size: 1.5rem;
      font-weight: 700;
      color: var(--text-title);
    }

    p {
      margin-top: 0.2rem;
    }

    div {
      display: flex;
      align-items: center;

      margin-top: 0.3rem;
      gap: 0.9rem;

      span.stargazers {
        display: flex;
        align-items: center;
      }

      span {
        font-size: 0.75rem;
        color: #58a6ff;
      }
    }
  }
`;
