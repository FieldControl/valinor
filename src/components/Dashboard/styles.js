import styled from "styled-components"

export const FirstContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin: 15px;
  font-size: 20px;
`

export const FullContainer = styled.div`
  margin: 10px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: flex-start;
`
export const StyleCard = styled.div`
  .cards {
    margin: 5px;
  }
`

export const DivChart = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
`

export const InsideCard = styled.div`
  display: flex;
`
