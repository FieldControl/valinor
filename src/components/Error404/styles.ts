/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';

export const Container = styled.div`
  height: calc(100% - 94px);

  #auth {
    position: absolute;
    top: 0;
    right: 0;
    z-index: 50;
    min-height: 32px;
    background-color: rgba(53, 95, 120, 0.4);
    padding: 7px 10px;
    border-bottom-right-radius: 10px;
    border-bottom-left-radius: 10px;
    box-shadow: 0 3px 0 rgba(0, 0, 0, 0.28);
    display: none;
  }
  #auth h1,
  #auth p,
  #auth label {
    display: none;
  }
  .auth-form-body {
    display: inline;
  }

  .logo {
    display: inline-block;
    margin-top: 35px;
  }
  .logo-img-2x {
    display: none;
  }
  @media only screen and (-webkit-min-device-pixel-ratio: 2),
    only screen and (min--moz-device-pixel-ratio: 2),
    only screen and (-o-min-device-pixel-ratio: 2/1),
    only screen and (min-device-pixel-ratio: 2),
    only screen and (min-resolution: 192dpi),
    only screen and (min-resolution: 2dppx) {
    .logo-img-1x {
      display: none;
    }
    .logo-img-2x {
      display: inline-block;
    }
  }
  #suggestions {
    margin-top: 35px;
    color: #ccc;
  }
  #suggestions a {
    color: #666666;
    font-weight: 200;
    font-size: 14px;
    margin: 0 10px;
  }

  #parallax_wrapper {
    position: relative;
    height: 100%;
    z-index: 0;
    -webkit-transition: all 0.25s ease-in;
    transition: all 0.25s ease-in;
  }
  #parallax_field {
    overflow: hidden;
    position: absolute;
    height: 100%;
    left: 0;
    top: 0;
    width: 100%;
  }
  #parallax_field #parallax_bg {
    position: absolute;
    width: 110%;
    height: 110%;
    z-index: 1;
  }
  #parallax_illustration {
    display: block;
    margin: 0 auto;
    width: 940px;
    height: 100%;
    position: relative;
    overflow: hidden;
    clear: both;
  }
  #parallax_illustration img {
    position: absolute;
  }
  #parallax_illustration #parallax_error_text {
    top: 272px;
    left: 72px;
    z-index: 10;
  }
  #parallax_illustration #parallax_octocat {
    top: 294px;
    left: 356px;
    z-index: 9;
  }
  #parallax_illustration #parallax_speeder {
    top: 350px;
    left: 432px;
    z-index: 8;
  }
  #parallax_illustration #parallax_octocatshadow {
    top: 497px;
    left: 371px;
    z-index: 7;
  }
  #parallax_illustration #parallax_speedershadow {
    top: 463px;
    left: 442px;
    z-index: 6;
  }
  #parallax_illustration #parallax_building_1 {
    top: 273px;
    left: 467px;
    z-index: 5;
  }
  #parallax_illustration #parallax_building_2 {
    top: 313px;
    left: 762px;
    z-index: 4;
  }
`;
