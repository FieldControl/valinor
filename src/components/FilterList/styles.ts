import styled, { css } from "styled-components";

interface IFilterButton {
  active: boolean;
}

export const Container = styled.div`
  margin-bottom: 16px;
  background-color: #0d1117;
  display: flex;
  flex-direction: column;
`;

export const Filter = styled.button<IFilterButton>`
  background-color: transparent;
  color: #c9d1d9;
  padding: 8px 16px;
  border: 2px solid #0d1117;
  border-bottom: 1px solid #21262d;
  display: flex;
  justify-content: space-between;
  font-size: 14px;
  min-width: 230px;

  &:first-child {
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
  }

  &:hover {
    background-color: #161b22;
    border-color: #161b22;
    cursor: pointer;
  }

  ${({ active }) =>
    active &&
    css`
      border-left-color: #f28065;
      border-top-color: #161b22;
      background-color: #161b22;
      border-top-left-radius: 0;
    `}
`;

export const Amount = styled.p`
  min-width: 20px;
  padding: 0 6px;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  background-color: #6e7681;
  color: #f0f6fc;
  border-radius: 2em;
`;
