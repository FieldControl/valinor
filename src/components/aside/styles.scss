.section-1,
.section-2 {
  border: 1px solid #2d333a;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  height: 400px;
  width: 220px;
  margin-top: 40px;
  margin-left: 261px;
  color: white;

  > button {
    color: white;
    display: flex;
    align-items: center;
    padding: 10px;
    width: 220px;
    height: 40px;
    font-size: 13px;
    font-weight: 200;
    background: none;
    outline: none;
    border: none;
    border-top: 1px solid #2d333a;

    &:hover {
      cursor: pointer;
      background-color: #2d333a65;
    }
    > span {
      color: white;

      display: flex;
      margin-left: auto;
      font-weight: bold;
      font-size: 11px;
      padding-left: 10px;
      padding-right: 10px;
      background-color: #6e7681;
      border-radius: 10px;
    }
  }
}

.section-2 {
  height: 263px;
  font-size: 12px;
  margin-top: 15px;
  padding-top: 10px;
  .title-section {
    display: flex;
    align-items: center;
    padding-left: 20px;
    padding-bottom: 10px;
    font-size: 14px;
    font-weight: 400;
  }

  > span {
    display: flex;
    align-items: center;
    padding: 10px 10px 10px 30px;
    color: #616b76;
    font-size: 10.4px;
  }
  span:hover {
    background-color: #2d333a65;
  }

  .search-advance {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    > a {
      font-size: 10px;
      margin-top: 30px;
      margin-right: 24px;
      color: #65a4ec;
      text-decoration: none;
      padding-right: 5px;

      &:hover {
        text-decoration: underline;
      }
    }
  }
}

@media (max-width: 1440px) {
  .section-1,
  .section-2 {
    display: none;
  }
}
