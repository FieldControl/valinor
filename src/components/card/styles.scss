.card {
  display: flex;
  color: #fff;
  position: relative;
  border-top: 1px solid #2d333a;
  width: 750px;
  margin-left: 580px;
  height: 120px;

  .name {
    display: flex;
    padding-top: 15px;
    font-size: 0.87rem;
    color: #2482ed;
    cursor: pointer;
    .book {
      margin: 1px 10px 0 0;
    }
  }

  .details-pokemon {
    margin-top: 20px;
    position: relative;
    font-size: 0.83rem;
    color: #81888f;
  }
  .types-container {
    display: flex;
    padding: 1px;
  }

  .list-types {
    display: flex;
    position: relative;
    top: 12px;
    right: 100px;
    .types {
      background-color: #13233a;
      color: #88b6f7;
      outline: none;
      border: none;
      margin: 50px 20px 0 0;
      height: 22px;
      width: 70px;
      padding-top: 5px;
      border-radius: 20px;
      font-size: 0.68rem;
      font-weight: 400;
      transition: 0.4s;
      cursor: pointer;
      &:hover {
        background-color: #2482ed;
        color: #fff;
        font-weight: 600;
        transform: scale(1.03);
      }
    }
  }
  .details-list {
    font-size: 0.82rem;
    position: absolute;
    margin-top: 95px;
    font-size: 10px;
    text-align: center;
    display: flex;
    color: #9fa7b1;

    .star {
      margin-right: 5px;
    }

    span {
      position: relative;
      top: 3px;
      padding-right: 20px;
    }
  }
}
@media (max-width: 1440px) {
  .card {
    width: 90vw;

    .details-pokemon {
      display: none;
    }
  }
}
