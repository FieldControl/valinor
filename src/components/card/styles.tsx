import styled from "styled-components";

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
    width: 60%;
    height: auto;
    padding: 16px;
    border-bottom: 1px solid #EAEFED;
    gap: .5rem;
`;

export const Topics = styled.div`
    display: flex;
    flex-wrap: wrap;
    gap: .5rem;

    button {
        display: flex;
        text-align: center;
        border: 1px solid transparent;
        background: #1f6feb;
        border-radius: 2em;
        color: #ffffff;
        padding: .2rem;
        cursor: pointer;
    };
`;

export const Info = styled.div`
    display: flex;
    gap: .5rem;
`;