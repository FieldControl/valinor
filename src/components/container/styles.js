import styled from 'styled-components';

export const Container = styled.main`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-left: 5%;
  padding-right: 5%;
`;
