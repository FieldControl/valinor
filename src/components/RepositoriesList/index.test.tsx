import { mount } from "enzyme";
import Repositories from ".";
import { Image } from "./styles";
import GlobalState from "../../global/GlobalState";

describe("Repositories", () => {
  it("should match the snapshot", () => {
    const wrapper = mount(
      <GlobalState>
        <Repositories />
      </GlobalState>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the octocat image", () => {
    const wrapper = mount(
      <GlobalState>
        <Repositories />
      </GlobalState>
    );
    expect(wrapper.find(Image)).toHaveLength(1);
  });
});
