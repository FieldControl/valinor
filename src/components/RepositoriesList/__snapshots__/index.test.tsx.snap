// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories should match the snapshot 1`] = `
<GlobalState>
  <RepositioriesList>
    <styled.div>
      <div
        className="sc-iqHYmW bpfviY"
      >
        <styled.div>
          <div
            className="sc-bqyKOL eZzUnP"
          >
            <styled.img
              src="octocat.png"
            >
              <img
                className="sc-crrszt fvrbCA"
                src="octocat.png"
              />
            </styled.img>
            <styled.h3>
              <h3
                className="sc-dQoVA eRYyiZ"
              >
                Faça a sua busca por repositórios preenchendo o campo acima
              </h3>
            </styled.h3>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </RepositioriesList>
</GlobalState>
`;
